service.subtitles.legendasdivx/service.py

```diff
@@ -0,0 +1,584 @@
+            if _parentfolder == '1':
+                if re.search(dirsearch[-1], desc, re.IGNORECASE): sync = True
+                else: sync = False
+            if _parentfolder == '2':
+                if (searchstring_notclean != ""):
+                    sync = False
+                    if str.lower(searchstring_notclean) in str.lower(desc.decode('utf8', 'ignore')): sync = True
+                else:
+                    if (str.lower(dirsearch_check[-1]) == "rar") or (str.lower(dirsearch_check[-1]) == "cd1") or (str.lower(dirsearch_check[-1]) == "cd2"):
+                        sync = False
+                        if len(dirsearch) > 1 and dirsearch[1] != '':
+                            if re.search(filesearch[1][:len(filesearch[1])-4], desc, re.IGNORECASE) or re.search(dirsearch[-2], desc, re.IGNORECASE): sync = True
+                        else:
+                            if re.search(filesearch[1][:len(filesearch[1])-4], desc, re.IGNORECASE): sync = True
+                    else:
+                        sync = False
+                        if len(dirsearch) > 1 and dirsearch[1] != '':
+                            if re.search(filesearch[1][:len(filesearch[1])-4], desc) or re.search(dirsearch[-1], desc, re.IGNORECASE): sync = True
+                        else:
+                            if re.search(filesearch[1][:len(filesearch[1])-4], desc, re.IGNORECASE): sync = True
+            if _filenameon == "false": filename = "From: " + uploader + " - "  + desc + "  " + "hits: " + hits
+            else: filename = "From: " + uploader + " - "  + filename + " " + "(" + movieyear + ")" + "  " + "hits: " + hits + " - " + desc
+            subtitles_list.append({'rating': str(downloads), 'filename': filename, 'uploader': uploader, 'desc': desc, 'sync': sync, 'hits' : hits, 'id': id, 'language_short': languageshort, 'language_name': languagelong})
+            log(u"getallsubs: SUBS LIST = '%s'" % subtitles_list)
+        page = page + 1
+        
+        if languageshort == "pt": content = urlpost(searchstring, "28", page)
+        elif languageshort == "pb": content = urlpost(searchstring, "29", page)
+        elif languageshort == "es": content = urlpost(searchstring, "30", page)
+        elif languageshort == "en": content = urlpost(searchstring, "31", page)
+
+#   Bubble sort, to put syncs on top
+    for n in range(0,len(subtitles_list)):
+        for i in range(1, len(subtitles_list)):
+            temp = subtitles_list[i]
+            if subtitles_list[i]["sync"] > subtitles_list[i-1]["sync"]:
+                subtitles_list[i] = subtitles_list[i-1]
+                subtitles_list[i-1] = temp
+    return subtitles_list
+
+def append_subtitle(item):
+    
+    listitem = xbmcgui.ListItem(
+                   label=item['language_name'],
+                   label2=item['filename']
+               )
+    listitem.setArt( { "icon": str(int(round(float(item["rating"])))), "thumb" : item["language_short"] } )
```

Comment:
is the icon correct? item["rating"] for an icon seems odd. Same for the thumb. I'm not familiar with subtitle addons though.

Reply:
It's good, don't worry :)